server/server.go

```diff
@@ -47,6 +52,69 @@ func socketConnectionHandler(pool types.Pool, w http.ResponseWriter, r *http.Req
 	go ws.Read()
 }
 
+func uploadImageHandler(w http.ResponseWriter, r *http.Request, db *types.ImageDB) {
+	log.Println("Upload Image endpoint hit!")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	switch r.Method {
+	case http.MethodPost:
+		// From https://socketloop.com/tutorials/upload-multiple-files-golang
+		err := r.ParseMultipartForm(200000) // grab the multipart form
+		if err != nil {
+			log.Println(err)
+			return
+		}
+
+		formdata := r.MultipartForm // ok, no problem so far, read the Form data
+
+		// Get the tag list from the form
+		tags := r.Form.Get("tag")
+		numfiles, _ := strconv.Atoi(r.Form.Get("length"))
+
+		tagsList := strings.Split(tags, ",")
+		for n := 0; n < numfiles; n++ { // loop through the files one by one
+			key := "files_" + strconv.Itoa(n)
+			formfile := formdata.File[key][0]
+			file, err := formfile.Open()
+			defer file.Close()
+			if err != nil {
+				fmt.Fprintln(w, err)
+				return
+			}
+
+			// Create a new UUID for the file, and get the file extention, and save to that file
+			newFileName := uuid.New().String()
+			extension := path.Ext(formfile.Filename)
+			filepath := "./assets/images/" + newFileName + extension
+			out, err := os.Create(filepath)
+
+			defer out.Close()
+			if err != nil {
+				fmt.Fprintf(w, "Unable to create the file for writing. Check your write access privilege")
+				return
+			}
+
+			_, err = io.Copy(out, file) // file not files[i] !
+
+			if err != nil {
+				fmt.Fprintln(w, err)
+				return
+			}
+
+			// If everything was successful, append the new image to the json file
+			db.Images = append(db.Images, types.TaggedImage{
+				Tags:     tagsList,
+				FilePath: filepath,
+			})
+			db.NumImages++
```

Comment:
Could these updates be handled internally to the database, like with a `db.AddImage()` function? I think we should try to avoid modifying the `db.Images` or `db.NumImages` fields within external files like `server.go`

Reply:
Good ideaz